src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ApplicationTime;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+            ->addSelect('context')
```

Comment:
When you have multiple selects you can add them all in a single statement `->addSelect(['context', 'promotionOfCoreEntity', 'promotionOfImage'])` this is what we do everywhere else.